core/src/main/java/org/elasticsearch/action/admin/indices/template/put/PutIndexTemplateRequest.java

```diff
@@ -449,7 +458,7 @@ public void readFrom(StreamInput in) throws IOException {
         super.readFrom(in);
         cause = in.readString();
         name = in.readString();
-        template = in.readString();
+        template = in.readList(StreamInput::readString);
```

Comment:
OK! New fun things for you to learn about. The way we support mixed version cluster (like, during a rolling restart) is by adding branches to this code. So this needs to read like:
```
if (in.getVersion().onOrAfter(Version.5_1_0)) {
  template = in.readList(StreamInput::readString);
} else {
  template = singletonList(in.readString());
}
```

And the `out` needs to have the same branching. And you need to find some place to cram the test for this.

The specific version number that I chose is based on my expectation of which release will be the first to see this feature.